src/NugetFeedDeploymentRepository/NugetFeedApplicationRepository.cs

```diff
@@ -0,0 +1,73 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Etg.Yams.Application;
+using Etg.Yams.Storage;
+using NuGet.Configuration;
+using NuGet.Packaging.Core;
+using NuGet.Protocol;
+using NuGet.Protocol.Core.Types;
+using NuGet.Versioning;
+
+namespace Etg.Yams.NuGet.Storage
+{
+    public class NugetFeedApplicationRepository : IApplicationRepository
+    {
+        private readonly SourceRepository _sourceRepository;
+
+        public NugetFeedApplicationRepository(string feedUrl = "https://api.nuget.org/v3/index.json", NugetFeedCredentials credentials = null)
+        {
+            this.FeedUrl = feedUrl;
+
+            List<Lazy<INuGetResourceProvider>> providers = new List<Lazy<INuGetResourceProvider>>();
+            providers.AddRange(Repository.Provider.GetCoreV3());
+            PackageSource packageSource = new PackageSource(feedUrl);
+
+            if(credentials != null)
+                packageSource.Credentials = new PackageSourceCredential("Yams Host", credentials.Username, credentials.Password, true);
+
+            _sourceRepository = new SourceRepository(packageSource, providers);
+        }
+
+        public string FeedUrl { get; private set; }
+
+        public Task DeleteApplicationBinaries(AppIdentity appIdentity)
+        {
+            throw new NotSupportedException("NuGet feeds are immutable.");
+        }
+
+        public async Task DownloadApplicationBinaries(AppIdentity appIdentity, string localPath, ConflictResolutionMode conflictResolutionMode)
+        {
+            string tempPath = Path.GetTempPath();
+            DownloadResource downloadResource = await _sourceRepository.GetResourceAsync<DownloadResource>();
+            PackageIdentity packageIdentity = new PackageIdentity(appIdentity.Id, new NuGetVersion(appIdentity.Version.ToString()));
+            PackageDownloadContext downloadContext = new PackageDownloadContext(new SourceCacheContext(), tempPath, true);
+
+            var result = await downloadResource.GetDownloadResourceResultAsync(packageIdentity, downloadContext, tempPath, new TraceLogger(), CancellationToken.None);
```

Comment:
Please only use var when return type is obvious

Reply:
Done